go/vt/vttablet/tabletserver/vstreamer/vstreamer.go

```diff
@@ -567,26 +571,28 @@ func (vs *vstreamer) parseEvent(ev mysql.BinlogEvent) ([]*binlogdatapb.VEvent, e
 		if err != nil {
 			return nil, err
 		}
-
 		if plan, ok := vs.plans[id]; ok {
 			// When the underlying mysql server restarts the table map can change.
 			// Usually the vstreamer will also error out when this happens, and vstreamer re-initializes its table map.
 			// But if the vstreamer is not aware of the restart, we could get an id that matches one in the cache, but
 			// is for a different table. We then invalidate and recompute the plan for this id.
-			if plan == nil || plan.Table.Name == tm.Name {
+			isInternal := tm.Database == sidecar.GetName()
+			if plan == nil ||
+				(plan.Table.Name == tm.Name && isInternal == plan.IsInternal) {
 				return nil, nil
 			}
 			vs.plans[id] = nil
 			log.Infof("table map changed: id %d for %s has changed to %s", id, plan.Table.Name, tm.Name)
 		}
 
-		if tm.Database == sidecar.GetName() && tm.Name == "resharding_journal" {
-			// A journal is a special case that generates a JOURNAL event.
-			return nil, vs.buildJournalPlan(id, tm)
-		} else if tm.Database == sidecar.GetName() && tm.Name == "schema_version" && !vs.se.SkipMetaCheck {
-			// Generates a Version event when it detects that a schema is stored in the schema_version table.
-			return nil, vs.buildVersionPlan(id, tm)
+		// The database connector `vs.cp` points to the keyspace's database.
+		// If this is also setup as the sidecar database name, as is the case in the distributed transaction unit tests,
+		// for example, we stream all tables as usual.
+		// If not, we only stream the schema_version and journal tables and those specified in the internal_tables list.
+		if tm.Database == sidecar.GetName() && vs.cp.DBName() != sidecar.GetName() {
+			return vs.buildSidecarTablePlan(id, tm)
```

Comment:
If we are no longer using buildJournalPlan and buildVersionPlan, we should delete them.

Reply:
fixed